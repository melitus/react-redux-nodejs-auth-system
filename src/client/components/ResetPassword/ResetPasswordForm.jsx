import React from 'react';
import { reduxForm } from 'redux-form';
import { Container, Grid, Form, Message } from 'semantic-ui-react';

import { Input } from '../../Field/index';
import ResetPasswordFormValidation from './ResetPasswordFormValidation';

const ResetPasswordForm = (props) => {
  const { handleSubmit, pristine, submitting, loading, error } = props;
  return (
    <Container>
      <Grid>
        <Grid.Row centered>
          <Grid.Column width={6}>
            <Form onSubmit={handleSubmit} autoComplete="off" loading={loading} >
            <label target="resetPasswordForm.password">Password</label>
              <Field
                id="resetPasswordForm.password"
                name="password"
                component={Input}
                placeholder="Password"
                type="password"
                value={password}
              />

              <label target="resetPasswordForm.confirmPassword">Confirm Password</label>
              <Field
                id="resetPasswordForm.confirmPassword"
                name=" confirmPassword"
                component={Input}
                placeholder="Confirm Password"
                type="password"
                value={ confirmPassword}
              />
              {error && (
                <Message negative>
                  <p>{error}</p>
                </Message>
                 )}
              <Form.Button primary type="resetpassword" disabled={pristine || submitting}>Reset Password</Form.Button>
            </Form>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};

export default reduxForm({
  form: 'ResetPasswordForm',
  validate: ResetPasswordFormValidation,
})(ResetPasswordForm);

