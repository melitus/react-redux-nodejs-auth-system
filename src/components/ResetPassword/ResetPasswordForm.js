import React from 'react';
import { reduxForm } from 'redux-form';
import { Container, Grid, Form, Message } from 'semantic-ui-react';

import { Input } from '../commons/Field/index';
import ResetPasswordFormValidation from './ResetPasswordFormValidation';

type ResetPasswordFormProp = {
  handleSubmit?: Function,
  loading?: boolean,
  pristine?: boolean,
  submitting?: boolean,
  error?: string
};

const ResetPasswordForm = ({handleSubmit, pristine, submitting, loading, error}:ResetPasswordFormProp) => {
  return (
    <Container>
      <Grid>
        <Grid.Row centered>
          <Grid.Column width={6}>
            <Form onSubmit={handleSubmit} autoComplete="off" loading={loading} >
            <label target="resetPasswordForm.password">Password</label>
              <Field
                id="resetPasswordForm.password"
                name="password"
                component={Input}
                placeholder="Password"
                type="password"
              />

              <label target="resetPasswordForm.confirmPassword">Confirm Password</label>
              <Field
                id="resetPasswordForm.confirmPassword"
                name=" confirmPassword"
                component={Input}
                placeholder="Confirm Password"
                type="password"
              />
              {error && (
                <Message negative>
                  <p>{error}</p>
                </Message>
                 )}
              <Form.Button primary type="resetpassword" disabled={pristine || submitting}>Reset Password</Form.Button>
            </Form>
          </Grid.Column>
        </Grid.Row>
      </Grid>
    </Container>
  );
};

export default reduxForm({
  form: 'ResetPasswordForm',
  validate: ResetPasswordFormValidation,
})(ResetPasswordForm);

