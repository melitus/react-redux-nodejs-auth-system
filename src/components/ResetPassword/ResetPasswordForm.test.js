import React from 'react';
import enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15.4';

import ResetPasswordForm from './ResetPasswordForm';

enzyme.configure({ adapter: new Adapter() });

describe('<ResetPasswordForm />', () => {
  it('should render default snapshot', () => {
    const wrapper = shallow(
      <ResetPasswordForm />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
