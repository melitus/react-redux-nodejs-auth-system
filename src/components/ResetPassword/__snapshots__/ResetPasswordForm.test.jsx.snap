exports[`<ResetPasswordForm /> should render default snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ReduxForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "destroyOnUnmount": true,
      "enableReinitialize": false,
      "forceUnregisterOnUnmount": false,
      "form": "ResetPasswordForm",
      "getFormState": [Function],
      "initialValues": undefined,
      "keepDirtyOnReinitialize": false,
      "persistentSubmitErrors": false,
      "pure": true,
      "shouldAsyncValidate": [Function],
      "shouldError": [Function],
      "shouldValidate": [Function],
      "shouldWarn": [Function],
      "touchOnBlur": true,
      "touchOnChange": false,
      "updateUnregisteredFields": false,
      "validate": [Function],
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "destroyOnUnmount": true,
        "enableReinitialize": false,
        "forceUnregisterOnUnmount": false,
        "form": "ResetPasswordForm",
        "getFormState": [Function],
        "initialValues": undefined,
        "keepDirtyOnReinitialize": false,
        "persistentSubmitErrors": false,
        "pure": true,
        "shouldAsyncValidate": [Function],
        "shouldError": [Function],
        "shouldValidate": [Function],
        "shouldWarn": [Function],
        "touchOnBlur": true,
        "touchOnChange": false,
        "updateUnregisteredFields": false,
        "validate": [Function],
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenFourAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
